### ISSUE TITLE
Ability to change the IMGIX path URL

### ISSUE THREAD

**Issue by sebastiangraz** [bigcartel/middleman-imgix#2]
Not sure if I overlooked something but when I generate an image I get a url that looks like this:

`<img ix-src="https://your-subdomain.imgix.net/images/example.png?ixlib=rb-1.1.0&auto=format" />
`

Which is fine but the `/images/` bit gets me because my Imgix domain does not have an `/images/` appended after the baseurl. 

is there a way to get this url instead? (without the extra /images/ path): 

`<img ix-src="https://your-subdomain.imgix.net/example.png?ixlib=rb-1.1.0&auto=format" />
`

**Comment by mattwigham** [comment]
That's likely middleman inserting that. Can you check that you're starting your image paths with a `/`? So `/example.png` instead of `example.png`.

**Comment by sebastiangraz** [comment]
Hi, thanks for the quick reply.

`<%= image_tag 'case-poetry.png' %>
`
Renders:

`https://grazio.imgix.net/images/case-poetry.png?ixlib=rb-1.1.0&auto=format&s=f40dbc887bbf2cbe270756bcf157649c`

However with the slash prepended like so

`<%= image_tag '/case-poetry.png' %>`

gives me 

`<img src="/case-poetry.png" alt="Case poetry">`

My setup looks like this: 

`
activate :imgix, host: 'grazio.imgix.net', secure_url_token: 'MYTOKEN', shard_strategy: :cycle
`

**Comment by mattwigham** [comment]
Interesting. What version of Middleman are you using?

**Comment by sebastiangraz** [comment]
```
gem 'middleman', '>= 4.0.0'
gem 'middleman-blog', "~> 4.0"
gem 'middleman-imgix'
```

**Comment by mattwigham** [comment]
So, now that I think about it, the path will just mimic the local path. So if your `images_dir` is set to `images` locally (which is the default), it'll just look for that path on your Imgix host. If you customize your [images_dir](http://www.rubydoc.info/gems/middleman-core/Middleman/Application#images_dir-instance_method) in config.rb it'll change the path on Imgix.

**Comment by sebastiangraz** [comment]
Ah yes, makes sense. Looks like the gem only accepts the default value of 'images' (`set :images_dir, 'images'`) at the moment. Changing it to 'img' broke the plugin and only renders: 

`<img src="/img/case-poetry.png" alt="Case poetry">`

I'm curious why the plugin is relying on `image_tag` helper rather than a custom helper? Seems like there is some collision happening. 

Also if I am to change `images_dir` to have no subdirectories? How do I do that? I want to be able to access imgix with just the grazio.imgix.net/**myimages.jpg**

Maybe it's my setup?

**Comment by mattwigham** [comment]
Can you make sure your images are actually in the `source/img` directory? Middleman won't send image paths to this plugin if the image doesn't actually exist at that location locally.

Also, this plugin doesn't actually rely on the `image_tag` plugin, it rewritings image URLs at the source, so that anywhere they're used (`image_tag`, sitemap, etc) it's pulling from the Imgix location.

**Comment by sebastiangraz** [comment]
Hi again. Apologies if this issue is a bit trivial, and perhaps a bit hard to debug — and yes the images are located locally. 

Here is a screenshot of the current sitemap:

![screen shot 2016-12-07 at 18 54 51](https://cloud.githubusercontent.com/assets/305057/20982027/c976f814-bcae-11e6-9ca9-64f2758733ec.png)

If it helps I have uploaded the whole project to a repo and can be found here. The image reference line of code can be found here: 

https://github.com/umbriel/portfolio-prototype/blob/master/source/index.html.erb#L11 

Perhaps theres a repo you may know of that has middleman-imgix successfully implemented that I can have a look at?

**Comment by mattwigham** [comment]
So I think the disconnect is you seem to want your images on Imgix in a different directory structure from what you have locally, and that won't work with this plugin. 

So if you want your images on Imgix to be in `/img` then you'd need to have all of your images locally in `source/img` and set your `images_dir` to `"img"` as well. Or, like we do, leave it in the default `source/images` and have Imgix just pull from that directory.

Long story short, your image paths on Imgix need to mirror your local ones.
